x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/CommonEqlActionTestCase.java

```diff
@@ -39,13 +46,20 @@ public static void checkForSnapshot() {
         assumeTrue("Only works on snapshot builds for now", Build.CURRENT.isSnapshot());
     }
 
-    @Before
+    private static boolean isSetUp = false;
```

Comment:
You could use the approach used in SQL:https://github.com/elastic/elasticsearch/blob/8279f82dea4d80e138841de1e9f8da724af4b313/x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SpecBaseIntegrationTestCase.java#L60, that is on each test check whether the index exists - if it doesn't load the data, otherwise ignore it.

And wipe the data in a `@AfterClass`. Note that you could do the loading inside `@BeforeClass` as well if there are no modifications to the index at all.

Reply:
The client from the base class only available on after @Before hook. Unless I hack around that, the safest bet is to initialize on @Before and since I need to initialize it once that's why synchronizing on the class